src/librustc_interface/passes.rs

```diff
@@ -568,14 +568,13 @@ fn write_out_deps(
             boxed_resolver.borrow().borrow_mut().access(|resolver| {
                 for cnum in resolver.cstore().crates_untracked() {
                     let source = resolver.cstore().crate_source_untracked(cnum);
-                    if let Some((path, _)) = source.dylib {
-                        files.push(escape_dep_filename(&FileName::Real(path)));
-                    }
-                    if let Some((path, _)) = source.rlib {
-                        files.push(escape_dep_filename(&FileName::Real(path)));
-                    }
-                    if let Some((path, _)) = source.rmeta {
-                        files.push(escape_dep_filename(&FileName::Real(path)));
+                    match source {
```

Comment:
Oh! I just reaized tha this change was incorrect. (At least, I assume we can have scenarios where the paths here are different, and we need to make multiples calls to `files.push`!) Sorry!

Reply:
(fixed by removing the erroneous refactoring and living with some resulting ugly code duplication.)